packages/react-vapor/src/components/sideNavigation/examples/SideNavigationExample.tsx

```diff
@@ -25,23 +26,40 @@ export class SideNavigationExample extends React.Component<any, any> {
                 <label className="form-control-label">Side Navigation</label>
                 <div className="flex flex-row flex-stretch">
                     <SideNavigation>
-                        <SideNavigationMenuSection header={{title: 'Section 1', svgName: 'identity-user'}}>
-                            <SideNavigationItem href="http://coveo.com" title="Link to Coveo" />
+                        <SideNavigationMenuSection title="Section 1" svgName="identity-user">
+                            <SideNavigationItem isActive href="http://coveo.com" title="Link to Coveo" />
                             <SideNavigationItem href="http://coveo.com" title="Another link to Coveo" />
                         </SideNavigationMenuSection>
-                        <SideNavigationMenuSection header={{title: 'Section 2'}}>
+                        <SideNavigationMenuSection title="Section 2">
                             <SideNavigationLoadingItem className="mod-width-30" />
                             <SideNavigationLoadingItem className="mod-width-50" />
                             <SideNavigationLoadingItem className="mod-width-40" />
                         </SideNavigationMenuSection>
                         <SideNavigationMenuSection
-                            header={{title: 'Section 3', svgName: 'menu-content'}}
+                            title="Collapsible Section"
+                            svgName="menu-content"
                             expandable
                             expanded={this.state.expanded}
                             onClick={() => this.click()}
                         >
-                            <SideNavigationItem href="http://coveo.com" target="_blank" title="Link to Coveo" />
-                            <SideNavigationItem href="http://coveo.com" title="Another link to Coveo" />
+                            <SideNavigationItem>

```

Comment:
To be quite honest, I think I prefer the old style.

No concern about "how", just fill in the props and the component will be rendered with the right style.

With the new way, I have to know that a `<a href>` goes inside and that I must put a `className="navigation-menu-section-item-link"`, which will be duplicated for all of my links.

If you really want to go the "super flexible" way, I think it would be better to also offer a `SideNavigationItemLink` that provides this full easy functionality

Reply:
I preserved the old way of doing things, because I agree it looks better, but the new way is required if you want to bind the navigation item to `react-router-dom` properly like I did in the demo:

```tsx
const NavLink: React.FunctionComponent<{href: string; name: string; baseUrl?: string}> = ({
    href,
    name,
    baseUrl,
    ...rest
}) => {
    const path = `${baseUrl}${href}`;
    return (
        <Route
            path={path}
            children={(routeProps: RouteChildrenProps) => (
                <SideNavigationItem isActive={!!routeProps.match}>
                    <Link to={path} className="block navigation-menu-section-item-link" {...rest}>
                        {name}
                    </Link>
                </SideNavigationItem>
            )}
        />
    );
};
```

`Link` already outputs an anchor tag, so if the SideNavigationItem does too by default you'll end-up having 2 embedded `<a>`;

Reply:
Maybe we could add the `className="navigation-menu-section-item-link"` automatically on the children?

Reply:
Done